lintcheck/src/main.rs

```diff
@@ -371,127 +402,15 @@ impl Crate {
             return Vec::new();
         }
 
-        let output_lines = stdout.lines();
-        let warnings: Vec<ClippyWarning> = output_lines
-            .into_iter()
-            // get all clippy warnings and ICEs
-            .filter(|line| filter_clippy_warnings(&line))
-            .map(|json_msg| parse_json_message(json_msg, &self))
+        // get all clippy warnings and ICEs
+        let warnings: Vec<ClippyWarning> = Message::parse_stream(stdout.as_bytes())
+            .filter_map(|msg| ClippyWarning::new(msg.unwrap(), &self))
             .collect();
 
```

Comment:
oh wow, so we can basically pipe rustc output into cargo-metadata and it will let us do some fancy processing of the stdout/stderr? that's super useful to know :grin:

Reply:
Yeah it's handy indeed

If you're parsing rustc output directly the format is slightly different from cargo's, the json is untagged and it'll emit [`Diagnostic`s](https://docs.rs/cargo_metadata/latest/cargo_metadata/diagnostic/struct.Diagnostic.html) directly